lib/model/query/entities.js

```diff
@@ -62,55 +62,49 @@ select ins.*, def.id as "entityDefId", ds."acteeId" as "dsActeeId", ds."name" as
-    LIMIT 1),
-  entity AS (
-    SELECT entities."datasetId", entities."uuid", entities."id" from entities
-    WHERE entities."uuid" = ${entityUuid}
-  )
-  select dataset."id" as "datasetId", entity."id" as "entityId" from dataset
-  FULL OUTER JOIN entity
-    ON dataset."id" = entity."datasetId"`)
-    .then(({ datasetId, entityId }) =>
-      (datasetId !== null && entityId == null));
 
 // Entrypoint to where submissions (a specific version) become entities
-const processSubmissionDef = (submissionDefId) => ({ Datasets, Entities, Submissions }) =>
-  new Promise((resolve, reject) => {
-    Entities.getDefBySubmissionDefId(submissionDefId)
-      .then((def) => (def.isDefined()
-        ? reject(Problem.user.entityViolation({ reason: 'This submission was already used to create an entity.' }))
-        : Submissions.getDefById(submissionDefId).then((s) => s.get())
-          .then((submissionDef) => Datasets.getFieldsByFormDefId(submissionDef.formDefId)
-            .then((fields) =>
-              ((fields.length === 0)
-                ? resolve()
-                : Entity.fromSubmissionXml(submissionDef.xml, fields)
-                  .then((partial) =>
-                    Entities.canInsertEntity(partial.aux.dataset, submissionDefId, partial.uuid)
-                      .then((okToInsert) =>
-                        (okToInsert
-                          ? Entities.createNew(partial, submissionDef)
-                            .then((entity) => resolve(entity))
-                            .catch((err) => reject(Problem.user.entityViolation({ reason: 'Entity could not be created', err: err.problemDetails })))
-                          : reject(Problem.user.entityViolation({ reason: 'It is not possible to create an entity from this data.' })))))
-                  .catch((problem) => reject(problem)))))));
-  });
+const _processSubmissionDef = (submissionDefId) => async ({ Datasets, Entities, Submissions }) => {
+  try {
+    const existingEntity = await Entities.getDefBySubmissionDefId(submissionDefId);
+    // If the submission has already been used to make an entity, don't try again
+    // and don't log it as an error.
+    if (existingEntity.isDefined())
+      return null;
+    const submissionDef = await Submissions.getDefById(submissionDefId).then((s) => s.get());
+    const fields = await Datasets.getFieldsByFormDefId(submissionDef.formDefId);
+    // No fields found for this formDefId means there is nothing entity-related to parse out
+    // of this submission. Even entity system properties like ID and dataset name would be here
+    // if the form def is about datasets and entities.
+    if (fields.length === 0)
+      return null;
+    const partial = await Entity.fromSubmissionXml(submissionDef.xml, fields);
+    const entity = await Entities.createNew(partial, submissionDef);
+    return entity;
+  } catch (problem) {
+    if (problem.problemCode === 409.14)
```

Comment:
Clarify with comment that these are all mainly entity validation errors

Reply:
I've switched to catching the errors more generically so this comment isn't relevant anymore.